### PR TITLE
Add mongodb_session resource and docs.

### LINKED ISSUE #5540: Add resources for Mongo database support
## Detailed Description
(i)Use the mongodb_session Chef InSpec audit resource to test json queries run against a Mongodb.

(ii)Use the mongodb_ conf Chef InSpec audit resource to test the contents of the configuration file for Mongodb, typically located at
/etc/mongod.conf
/etc/mongos.conf

## Context
Needed for CIS benchmark and customer needs.

### PR DESCRIPTION
**Pull request by Vasu1105**
Signed-off-by: Vasu1105 <vasundhara.jagdale@chef.io>

## Description

Adds mongodb_session resource.
Adds documentation for the same.

It also includes a minor change in mongodb_conf resource docs.

## Related Issue
Another part of Fix #5540 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New content (non-breaking change)
- [ ] Breaking change (a content change which would break existing functionality or processes)

## Checklist:

- [ ] I have read the **CONTRIBUTING** document.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -20,6 +20,8 @@ end
 # but our runtime dep is still 3.9+
 gem "rspec", ">= 3.10"
 
+gem "mongo"
```
Comment (clintoncwolfe) [inline Gemfile]: We need to include the gem in the inspec.gemspec file, not in the Gemfile.

File: docs-chef-io/content/inspec/resources/mongodb_session.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+where
+
+- `mongodb_session` declares a user, password and database, connecting locally, with permission to run the query.
+- `query` contains the query to be run.
+- `its("params") { should eq(/expected-result/) }` compares the results of the query against the expected result in the test
+
+### Optional Parameters
+
+`mongodb_session` InSpec resource accepts `user`, `password`, `host`, `port`, `auth_source`, `auth_mech`, `ssl`, `ssl_cert`, `ssl_ca_cert`, `auth_mech_properties`.
```
Comment (clintoncwolfe) [inline docs-chef-io/content/inspec/resources/mongodb_session.md]: Several of these options are not documented.

Comment (Vasu1105) [reply docs-chef-io/content/inspec/resources/mongodb_session.md]: Thanks, @clintoncwolfe. I have updated the docs.

**Review by clintoncwolfe — APPROVED** [review]
Looks good to me, needs a doc team review

**Comment by Vasu1105** [comment]
@inspec/docs-team this is waiting for your review.

File: docs-chef-io/content/inspec/resources/mongodb_session.md [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+## Availability
+
+### Installation
+
+This resource is distributed along with Chef InSpec itself. You can use it automatically.
+
+## Syntax
+
+A `mongodb_session` resource block declares the `user`, `password`, 'database' to use for the session, and then the command to be run:
```
Comment (IanMadd) [inline docs-chef-io/content/inspec/resources/mongodb_session.md]:
```suggestion
A `mongodb_session` resource block declares the `user`, `password`, and `database` to use for the session and then the command to be run:
```

File: docs-chef-io/content/inspec/resources/mongodb_session.md [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+
+  describe mongodb_session(user: "username", password: "password", database: "test").query(key: value) do
+    its("params") { should match(/expected-result/) }
+  end
+
+where
+
+- `mongodb_session` declares a user, password and database, connecting locally, with permission to run the query.
```
Comment (IanMadd) [inline docs-chef-io/content/inspec/resources/mongodb_session.md]:
```suggestion
- `mongodb_session` declares a user, password, and database, connecting locally, with permission to run the query.
```

File: docs-chef-io/content/inspec/resources/mongodb_session.md [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+- `query` contains the query to be run.
+- `its("params") { should eq(/expected-result/) }` compares the results of the query against the expected result in the test
+
+### Optional Parameters
+
+`mongodb_session` InSpec resource accepts `user`, `password`, `host`, `port`, `auth_source`, `auth_mech`, `ssl`, `ssl_cert`, `ssl_ca_cert`, `auth_mech_properties`.
```
Comment (IanMadd) [inline docs-chef-io/content/inspec/resources/mongodb_session.md]:
```suggestion
The `mongodb_session` InSpec resource accepts `user`, `password`, `host`, `port`, `auth_source`, `auth_mech`, `ssl`, `ssl_cert`, `ssl_ca_cert`, and `auth_mech_properties` parameters.
```

File: docs-chef-io/content/inspec/resources/mongodb_session.md [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+
+In Particular:
+
+#### `host`
+
+Defaults to `127.0.0.1`
```
Comment (IanMadd) [inline docs-chef-io/content/inspec/resources/mongodb_session.md]:
```suggestion
The server host IP address. Default value: `127.0.0.1`.
```

File: docs-chef-io/content/inspec/resources/mongodb_session.md [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+
+#### `port`
+
+Defaults to `27017`
```
Comment (IanMadd) [inline docs-chef-io/content/inspec/resources/mongodb_session.md]:
```suggestion
The server port. Default value: `27017`.
```

File: docs-chef-io/content/inspec/resources/mongodb_session.md [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+
+#### `auth_mech`
+
+Defaults to `:scram`. The available opitions are `:scram256`, `:mongodb_x509`, `:aws`. Refer this [docs](https://docs.mongodb.com/ruby-driver/master/tutorials/ruby-driver-authentication/) for more understanding about these options.
```
Comment (IanMadd) [inline docs-chef-io/content/inspec/resources/mongodb_session.md]:
```suggestion
The authentication mechanism. The available options are: `:scram`, `:scram256`, `:mongodb_x509`, and `:aws`. Default value: `:scram`.

See the MongoDB documentation on [Ruby driver authentication](https://docs.mongodb.com/ruby-driver/current/reference/authentication/) for more information.
```

File: docs-chef-io/content/inspec/resources/mongodb_session.md [hunk 9]
```diff
@@ -0,0 +1,112 @@
...
+
+#### `auth_source`
+
+Defaults to given database name. `database` name is mandatory.
```
Comment (IanMadd) [inline docs-chef-io/content/inspec/resources/mongodb_session.md]:
```suggestion
The database where the user’s authentication credentials are stored. The default value is the database name that is passed as a parameter to the resource.
```

File: docs-chef-io/content/inspec/resources/mongodb_session.md [hunk 10]
```diff
@@ -0,0 +1,112 @@
...
+
+#### `ssl`
+
+Defaults to false. Set `true ` to use ssl transport. For ssl realted options also refer to this [docs](https://docs.mongodb.com/ruby-driver/master/tutorials/ruby-driver-authentication/#client-certificate-x-509) for more understanding.
```
Comment (IanMadd) [inline docs-chef-io/content/inspec/resources/mongodb_session.md]:
```suggestion
Whether to use the SSL security protocol or not. Set to `true` to use SSL transport, default value: `false`.  See the MongoDB documentation on [Ruby Driver authentication](https://docs.mongodb.com/ruby-driver/current/reference/authentication/#client-certificate-x-509) for more information.
```

File: docs-chef-io/content/inspec/resources/mongodb_session.md [hunk 11]
```diff
@@ -0,0 +1,112 @@
...
+
+Path to ssl certificate file.
+
+#### `ssl_ca_cert`
+
+Path to ssl ca cert file.
+
+#### `ssl_key`
+
+Path to ssl key file.
```
Comment (IanMadd) [inline docs-chef-io/content/inspec/resources/mongodb_session.md]:
```suggestion
Path to the SSL certificate file.

#### `ssl_ca_cert`

Path to the SSL Certificate Authority (CA) certificate file.

#### `ssl_key`

Path to SSL key file.
```

File: docs-chef-io/content/inspec/resources/mongodb_session.md [hunk 12]
```diff
@@ -0,0 +1,112 @@
...
+
+#### `auth_mech_properties`
+
+This accepts hash of authetication mechanism properties. This option is generally used with `aws` auth mechanism. Example of this is given in this docs [here](https://docs.mongodb.com/ruby-driver/master/tutorials/ruby-driver-authentication/#aws)
```
Comment (IanMadd) [inline docs-chef-io/content/inspec/resources/mongodb_session.md]:
```suggestion
A hash of the authentication mechanism properties. This option is generally used with the AWS authentication mechanism. See the MongoDB documentation on [Ruby Driver authentication using AWS](https://docs.mongodb.com/ruby-driver/current/reference/authentication/#aws) for more information.
```

File: docs-chef-io/content/inspec/resources/mongodb_session.md [hunk 13]
```diff
@@ -0,0 +1,112 @@
...
+
+### MongodDB query reference docs
+
+This resource is using mongo ruby driver to fetch the data.
+[MongoDB Ruby Driver authentication](https://docs.mongodb.com/ruby-driver/master/tutorials/ruby-driver-authentication/)
+
```
Comment (IanMadd) [inline docs-chef-io/content/inspec/resources/mongodb_session.md]:
```suggestion
This resource uses the [MongoDB Ruby Driver](https://docs.mongodb.com/ruby-driver/current/reference/authentication/) to fetch the data.

```

File: docs-chef-io/content/inspec/resources/mongodb_session.md [hunk 14]
```diff
@@ -0,0 +1,112 @@
...
+#### `ssl_key`
+
+Path to ssl key file.
+
+#### `auth_mech_properties`
+
+This accepts hash of authetication mechanism properties. This option is generally used with `aws` auth mechanism. Example of this is given in this docs [here](https://docs.mongodb.com/ruby-driver/master/tutorials/ruby-driver-authentication/#aws)
+
+### MongodDB query reference docs
```
Comment (IanMadd) [inline docs-chef-io/content/inspec/resources/mongodb_session.md]:
```suggestion
### MongodDB Query Reference Documentation
```

File: docs-chef-io/content/inspec/resources/mongodb_session.md [hunk 15]
```diff
@@ -0,0 +1,112 @@
...
+
+This resource is using mongo ruby driver to fetch the data.
+[MongoDB Ruby Driver authentication](https://docs.mongodb.com/ruby-driver/master/tutorials/ruby-driver-authentication/)
+
+## Examples
+
+The following examples show how to use this Chef InSpec audit resource.
+
+### Test the roles information using rolesInfo command of MongoDB
```
Comment (IanMadd) [inline docs-chef-io/content/inspec/resources/mongodb_session.md]:
```suggestion
### Test the roles information using the `rolesInfo` command in MongoDB.
```

File: docs-chef-io/content/inspec/resources/mongodb_session.md [hunk 16]
```diff
@@ -0,0 +1,112 @@
...
+      its(["role"]) { should eq "dbAdmin" }
+    end
+
+### Test the MongoDB user role.
+
+    describe mongodb_session(user: "foo", password: "bar", database: "test").query(usersInfo: "foo").params["users"].first["roles"].first do
+      its(["role"]) { should eq "readWrite" }
+    end
+
+### Test the params
```
Comment (IanMadd) [inline docs-chef-io/content/inspec/resources/mongodb_session.md]:
```suggestion
### Test the database parameters.
```

**Comment by Vasu1105** [comment]
Thanks @IanMadd. Updated the docs.

**Pull request merged**